### PR TITLE
Fix stalling in Pipeline command

### PR DESCRIPTION
**Pull request by astaric**
When a command (other than the first) in a pipeline wrote more than 64k to the stderr, and the output was consumed with the iter_lines function, the whole pipeline stalled. Fixed by reading the output of all commands where either stdout or stderr was set to PIPE.

This is the same issue as described in
https://github.com/tomerfiliba/plumbum/issues/548

Same problem still exists if command is executed with run(), but that code calls proc.communicate to read stderr/stdout and I do not know how to fix that.

### CODE REVIEW

File: tests/test_pipelines.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+from typing import List, Tuple
+
+import env
```
Comment (henryiii) [inline tests/test_pipelines.py]:
```suggestion
```

**Comment by vient** [comment]
@henryiii do you plan to release new version soon? Stalling pipelines is something we've bumped in several times, this fix seems important.

I don't know how new releases are made, but if there is anything I can help with, I am ready :)

**Comment by henryiii** [comment]
Yeah, I'm planning one pretty soon. I want to get a release out then drop Python 3.6.

**Comment by desertfury** [comment]
Hi @henryiii !
I wonder have you shipped release that contains this particular fix yet? Thanks!

**Pull request merged**